python/BioSimSpace/FreeEnergy/_relative.py

```diff
@@ -696,6 +1278,159 @@ def _difference(self, pmf_ref):
+
+        return (message)
+
+    def _check_overlap(self):
+        """Check the overlap of an FEP leg. 
+
+           Parameters
+           ----------
+
+           Returns
+           -------
+
+           Message - gives a warning if the overlap is less than 0.03 for any off-diagonal.
+
+
+        """
+
+        # Calculate the overlap for this object.
+        _, overlap = self.analyse()
+
+        # Now call the staticmethod passing in the overlap and the work_dir of the run.
+        return Relative.check_overlap(overlap, estimator=self._estimator)
+
+    @staticmethod
+    def plot(overlap, estimator=None, work_dir=None):
+        """Plot either the overlap or the dhdl of the transformation. 
+
+           Parameters
+           ----------
+
+           overlap : [ [ float, float, ... ] ], numpy.matrix
+               For MBAR, the overlap matrix. This gives the overlap between lambda windows.
+               For TI, the dhdl.
+
+           estimator : str
+               The estimator ('MBAR' or 'TI') used.
+
+           work_dir : str
+               The working directory for the free-energy perturbation
+               simulation.
+
+           Returns
+           -------
+
+           Overlap matrix plot or dhdl graph.
+
+        """
+        
+        # estimator must be MBAR for overlap matrix or TI for dhdl plot.
+        if estimator not in ['MBAR', 'TI', None]:
+            raise ValueError("'estimator' must be 'MBAR' or 'TI'. If 'None, data type will be inferred.")
+
+        if estimator is None:
+            if isinstance(overlap, _np.matrix):
+                estimator = "MBAR"
+            elif isinstance(overlap, _alchemlyb.estimators.ti_.TI) or isinstance(overlap, list):
+                estimator = "TI"
+            else:
+                raise TypeError("Data type for estimator = 'None' could not be inferred / does not match allowed data types.")
+
+        if estimator == "MBAR":
+            if not isinstance(overlap, _np.matrix):
+                        raise TypeError("'overlap' must be of type 'numpy.matrix' for 'MBAR'.\
+                            This is obtained from running analysis using estimator='MBAR'.")
+
+            # use the alchemlyb functionality to plot the overlap matrix
+            ax = _plot_mbar_overlap_matrix(overlap)
+            ax.set_title(f"overlap matrix")
+
+            if work_dir is not None:
+                ax.figure.savefig(
+                    f"{work_dir}/overlap_MBAR.png", bbox_inches='tight', pad_inches=0.0)      
+
+        elif estimator == 'TI':
+            if not isinstance(overlap, _alchemlyb.estimators.ti_.TI):
+                if not isinstance(overlap, list):
+                        raise TypeError("'overlap' must be of type 'alchemlyb.estimators.ti_.TI' for 'TI'.\
+                                          This is obtained from running analysis using estimator='TI'.\
+                                          Alternatively, it may also be of type 'list'.")
+
```

Comment:
Referring back to my earlier comment, I can now see what type `overlap` will be in the case of a TI estimator. I think it would be best to discuss ways to get around this issue, since the use of `overlap` is likely to lead to confusion here. I see that the output can be of type `_alchemlyb.estimators.ti_.TI`, or `list`. For the latter, you'd probably also need to validate the contents of the list, since Python lists can contain anything.